lib/client.js

```
@@ -1,17 +1,59 @@
 'use strict';
 
 const Uuid = require('uuid/v4');
+const Joi = require('@hapi/joi');
 
 class Client {
 
     constructor(transport, idGenerator = Uuid) {
 
         this.transport = transport;
         this.idGenerator = idGenerator;
+        this.validateMap = new Map();
+    }
+
+    addPrePublishValidator(stream, eventType, validationSchema) {
+
+        if (!Joi.isSchema(validationSchema)) {
+            throw new Error('Invalid Joi schema');
+        }
+
+        const streamAndEventType = `${stream}:${eventType}`;
+        this.validateMap.set(streamAndEventType, validationSchema);
+    }
+
+    validate(stream, eventType, data) {
+
+        const streamAndEventType = `${stream}:${eventType}`;
+        const streamEventTypeSchema = this.validateMap.get(streamAndEventType);
+
+        if (streamEventTypeSchema) {
+            const validationResult = streamEventTypeSchema.validate(data, { abortEarly: true });
+
+            if (validationResult.error) {
+                return false;
```

we should throw the error here - so we don't lost the `Joi` context + message about how/what failed

---

Thanks, updated.